### DISCUSSION TITLE
PNG to DDS dxt5: always getting 8bit DXT1

### DISCUSSION THREAD

**Discussion by hbeni** [ImageMagick/ImageMagick#3123] [Help] [answered]
Hello,
I currently try to create dxt5 images but i only ever get dxt1...

Source image:
```
$ magick identify 3138120.png
3138120.png PNG 16384x8192 16384x8192+0+0 8-bit sRGB 222.73MiB 3.970u 0:03.970

$ file 3138120.png
3138120.png: PNG image data, 16384 x 8192, 8-bit/color RGB, non-interlaced
```

I tried:
- `magick 3138120.png -define dds:compression=dxt5 dxt5.dds`
- `magick 3138120.png -define dds:compression=DXT5 dxt5.dds`
- `magick 3138120.png -define dds:compression=dxt5 -format dxt5 dxt5.dds`
- `magick 3138120.png -define dds:compression=DXT5 -format dxt5 dxt5.dds`
- `magick 3138120.png -define dds:compression=dxt5 -format DXT5 dxt5.dds`
- `magick 3138120.png -define dds:compression=DXT5 -format DXT5 dxt5.dds`
- `magick 3138120.png -define dds:compression=DXT5 dxt5:dxt5.dds`

Everytime with the result:
```
$ file dxt5.dds
dxt5.dds: Microsoft DirectDraw Surface (DDS): 16384 x 8192, compressed using DXT1
$ magick identify dxt5.dds
dxt5.dds DDS 16384x8192 16384x8192+0+0 8-bit sRGB 85.3335MiB 1.530u 0:01.531
```

I am on debian bullseye and installed with 1-2.dev/imei:
```
$ magick -version
Version: ImageMagick 7.0.10-57 Q16 x86_64 2021-01-14 https://imagemagick.org
Copyright: © 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lcms lqr lzma openexr pangocairo png tiff webp x xml zlib
```

**Comment by dlemstra** [comment, marked as answer]
The compression is changed from DXT5 to DXT1 when the image has no alpha channel.

**Reply by hbeni** [reply]
Ah! That explains it.
Are the other DXT5 properties written nonetheless?

**Reply by dlemstra** [reply]
Not sure what you are asking?

**Reply by hbeni** [reply]
Shouldn't the dxt5 file take little more space than the dxt1 one, regardless of compression?

**Reply by dlemstra** [reply]
You are not writing a DTX5 file when the compression is DTX1.

**Reply by hbeni** [reply]
So, this implies, that the selectable output type is depending on the input file then?

Is this the proper syntax: `magick 3138120.png -define dds:compression=dxt5 dxt5.dds`
- When the input file suites dxt5, it will write dxt5?
- when not, dxt1 is written?

**Reply by dlemstra** [reply]
That is correct.

**Reply by hbeni** [reply]
Thank you very much for clarifying this, thank you!
Also for responding so quick, that helped me alot!
